src/path.jl

```diff
@@ -118,7 +118,7 @@ path components
 
 Returns whether there is a parent directory component to the supplied path.
 """
-hasparent(fp::AbstractPath) = length(fp.segments) > 1
+hasparent(fp::AbstractPath) = length(fp.segments) > Int(isempty(fp.root))
```

Comment:
```suggestion
hasparent(fp::AbstractPath) = length(fp.segments) > isempty(fp.root)
```
Though that might be less clear, it is legal:

```
julia> 1 > false
true

julia> 2 > false
true

julia> 0 > true
false
```

Alt be more clear explict

```suggestion
hasparent(fp::AbstractPath) = length(fp.segments) > (isempty(fp.root) ? 0 : 1)
```

Maybe:

```suggestion
hasparent(fp::AbstractPath) = length(fp.segments) - isempty(fp.root) > 0
```

Reply:
I think I like the first one as long as that behaviour isn't likely to change.

Reply:
`Bool <: Integer` too late to change that now